applications/crossbar/src/cb_context.erl

```diff
@@ -800,62 +800,140 @@ find_schema(<<_/binary>> = Schema) ->
 %%--------------------------------------------------------------------
 -spec add_system_error(atom() | binary(), context()) -> context().
 add_system_error('too_many_requests', Context) ->
-    crossbar_util:response('fatal', <<"too many requests">>, 429, Context);
+    build_system_error(429, 'too_many_requests', <<"too many requests">>, Context);
 add_system_error('no_credit', Context) ->
-    crossbar_util:response('error', <<"not enough credit to perform action">>, 402, Context);
+    build_system_error(402, 'no_credit', <<"not enough credit to perform action">>, Context);
 add_system_error('unspecified_fault', Context) ->
-    crossbar_util:response('fatal', <<"unspecified fault">>, Context);
+    build_system_error(500, 'unspecified_fault', <<"unspecified fault">>, Context);
 add_system_error('account_cant_create_tree', Context) ->
-    crossbar_util:response('fatal', <<"account creation fault">>, Context);
+    build_system_error(500, 'account_cant_create_tree', <<"account creation fault">>, Context);
 add_system_error('account_has_descendants', Context) ->
-    crossbar_util:response('fatal', <<"account has descendants">>, Context);
+    build_system_error(500, 'account_has_descendants', <<"account has descendants">>, Context);
 add_system_error('faulty_request', Context) ->
-    crossbar_util:response_faulty_request(Context);
+    build_system_error(404, 'faulty_request', <<"faulty request">>, Context);
```

Comment:
I feel like "faulty_request" should be 400. 404 is pretty specific to not finding the "thing".